### PR TITLE
Homework27: Andrei Kulbitski

### PR DESCRIPTION
**Pull request by kulbitski-andrei**
Homework 27: positive and negative API tests for "https://alexqa.netlify.app/.netlify"

### CODE REVIEW

File: homework_27/service_methods/service_methods.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    to a randomly selected domain from a predefined list.
+
+    Returns:
+        str: A randomly generated email address.
+    """
+    domains = ["example.com", "mail.com", "test.com"]
+    letters = string.ascii_lowercase
+    username = ''.join(random.choice(letters) for _ in range(10))
+    domain = random.choice(domains)
+    email = f"{username}@{domain}"
```
Comment (Stml89) [inline homework_27/service_methods/service_methods.py:22]: More pytonic way

```suggestion
def random_char(char_num):
       return ''.join(random.choice(string.ascii_letters) for _ in range(char_num))

print (random_char(7)+"@gmail.com")
```

File: homework_27/tests/tests_negative.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        "phoneNumber": "+12345678901",
+        "address": "123 Main St",
+        "role": "user",
+        "referralCode": "ABCDEFGH"
+    }
+
+    response = requests.post(url, headers=headers, json=data, timeout=10)
+    assert response.status_code == 200
+    response = requests.post(url, headers=headers, json=data, timeout=10)
+    assert response.status_code == 409
```
Comment (Stml89) [inline homework_27/tests/tests_negative.py:38]: how about moving `post`, `get` and other methods to a separate module?

File: homework_27/test_data/variables.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+"""VARIABLES FOR API TESTS"""
+
+url = "https://alexqa.netlify.app/.netlify"
+token = ("eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9."
+         "eyJ1c2VySWQiOiIxMTY4NzI0NzY5MjY4MDM1N"
+         "jM4MDIiLCJpYXQiOjE3MjM4ODE0MzcsImV4cC"
+         "I6MTcyMzg4NTAzN30.KadleyNmrDry_1bq8aV"
+         "jmyRlca66j-0XaDCO3p89_HY")
```
Comment (Stml89) [inline homework_27/test_data/variables.py:8]: Not really save to store secret information in the repository

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework_27/service_methods/service_methods.py (hunk 1 above)
Comment (kulbitski-andrei) [reply homework_27/service_methods/service_methods.py:22]: Fixed.

File: homework_27/test_data/variables.py (hunk 3 above)
Comment (kulbitski-andrei) [reply homework_27/test_data/variables.py:8]: Fixed. Now token stored in config.json file and this file is added to .gitignore.
Also added get_token function that retrieves token from that file.
File structure:
{
    "api_token": " ... "
}

**Review by Stml89 — APPROVED** [review]

**Pull request merged**